usort/sorting.py

```diff
@@ -110,65 +111,72 @@ def partition_leading_lines(
 
 
 def fixup_whitespace(
-    initial_blank: Sequence[cst.EmptyLine], imports: List[SortableImport]
+    initial_blank: Sequence[str], imports: List[SortableImport]
 ) -> List[SortableImport]:
+    """
+    Normalize whitespace/comments on a block of imports before transforming back to CST.
+    """
     cur_category = None
     # TODO if they've already been reshuffled, there may have been a blank
     # (separator) line between a non-block and the first import, that's now in
     # the middle.
-    for i in imports:
-        old_blanks, old_comments = partition_leading_lines(i.node.leading_lines)
+    for imp in imports:
+        _old_blanks, old_comments = partition_leading_lines(imp.comments.before)
 
         if cur_category is None:
             blanks = initial_blank
-        elif i.sort_key.category_index != cur_category:
-            blanks = (cst.EmptyLine(),)
-        elif old_comments:
```

Comment:
This will take

```py
import a
#comment
import b
```

and insert a line before the comment when `a` and `b` are same category.  I'm not sure I see that feature preserved?

Reply:
I made some changes elsewhere that preserves existing line breaks with the comments, which passes the unit tests, and I think is better in spirit with trying to minimize actual formatting changes needed to sort imports. If the original code didn't have a blank line, then I think it's fine not adding one ourselves (assuming it's not on a category boundary), and just letting black do the work to enforce blank lines before comments.